Custom theme

taitus opened the pull request:
## Objectives

Update styles to match the JCyL style guide

## Visual Changes
- Home
![Home](https://github.com/rockandror/participacyl/assets/16189/daf17841-9442-4bc5-856d-1ce378ab1bbf)

- Login
![InicioDeSesion](https://github.com/rockandror/participacyl/assets/16189/e4ce9506-1270-4238-aeec-5bc14d2d9838)

- Colaborative Legislation index
![LegislacionColaborativa](https://github.com/rockandror/participacyl/assets/16189/1ea0891f-5278-4268-a9fc-dc8659f0d610)

- Colaborative Legislation proces
![LegislacionColaborativa-Proceso](https://github.com/rockandror/participacyl/assets/16189/d97c38a2-ff03-4540-9974-bf176c7b397a)

- Colaborative Legislation proces 2
![LegislacionColaborativa-Proceso2](https://github.com/rockandror/participacyl/assets/16189/32e7b221-f827-410a-859b-1be9dd7b0c5b)

- Colaborative Legislation proces 3
![LegislacionColaborativa-Proceso-crearPropuesta](https://github.com/rockandror/participacyl/assets/16189/418db2f9-e003-4266-8e4a-07af93448150)

- Help page
![PaginaDeAyuda](https://github.com/rockandror/participacyl/assets/16189/31044afa-97cd-4bae-8896-74c04562584c)


## Pending
- Update PNG logo
- Update Hightlight color

taitus committed:
Add logo for jcyl

We need to ask the JCyL for a .png logo.

taitus committed:
Override brand and black color

taitus committed:
Add brand color to foundation-palette

taitus committed:
Update public header styles to match the JCyL style guide

taitus committed:
Override background footer color

taitus committed:
Add hover links in footer with brand color

taitus committed:
Add brand color to subnavigation bar

taitus committed:
Update legislation styles to match the JCyL style guide

javierm commented on app/views/custom/layouts/_header.html.erb:
```diff
@@ -0,0 +1,52 @@
+
+    <h1 class="top-bar-title">
+      <%= link_to root_path, accesskey: "0" do %>
+        <%= image_tag(image_path_for("logo_header_jcyl.png"), alt: setting["org_name"]) %>
```
Maybe we can rename this file to ` logo_header.png` so we don't have to customize this code :thinking:.

I know we'll get an error if we do so :relieved:. There's a workaround mentioned in Consul 1.5.0 release notes:

> Due to an upgrade of the sprockets gem, if you're overwriting images under the app/assets/images/ folder by adding an image with the same name to the app/assets/images/custom/ folder, you might run into a Sprockets::DoubleLinkError (see issue https://github.com/consul/consul/issues/4882). As a workaround, you can delete or rename the image under app/assets/images/ which has the conflicting name.

Then we could remove both this file and the `devise` file :tada:.

javierm commented on app/assets/stylesheets/_consul_custom_overrides.scss:
```diff
@@ -10,3 +10,14 @@
+  secondary: #767676,
+  success: #3adb76,
+  warning: #ffae00,
+  alert: darken(#cc4b37, 1%),
```
Why darken? :thinking:

If there's a good reason, could you add it to the commit message? :pray:

javierm commented on app/assets/stylesheets/custom.scss:
```diff
@@ -2,3 +2,15 @@
 // folder and they will be loaded automatically.
+
+
+.public > .wrapper > header {
```
Just curious; why is this change in `custom.scss` while the other ones are in `custom/layout.scss`? The original styles are all in the same file :thinking:.

javierm commented on app/assets/stylesheets/custom/layout.scss:
```diff
@@ -0,0 +1,63 @@
+      color: $white;
+    }
+  }
+}
```
IMHO we can remove these changes. The default color (white) is neutral and goes well with any color scheme.

javierm commented on app/assets/stylesheets/custom/legislation.scss:
```diff
@@ -0,0 +1,11 @@
+      color: $brand;
+    }
+  }
+}
```
I think we can remove this change :thinking:. Consul by default keeps these links black when desktop mouse users hover, and AFAIK there aren't any plans to change that :thinking:.

javierm commented on app/assets/stylesheets/custom/participation.scss:
```diff
@@ -0,0 +1,14 @@
+      }
+    }
+  }
+}
```
Just like in other titles, I think we can remove this change :thinking:. Consul by default keeps these links black when desktop mouse users hover, and AFAIK there aren't any plans to change that :thinking:.

javierm commented on app/assets/stylesheets/custom/legislation_process.scss:
```diff
@@ -0,0 +1,30 @@
+    margin: rem-calc(-1) 0;
+    min-height: $line-height * 3;
+    position: relative;
+    vertical-align: top;
```
These styles are already in the original file, right? Can we remove them from here, then, or am I missing something? :thinking:

javierm commented on app/assets/stylesheets/custom/legislation_process.scss:
```diff
@@ -0,0 +1,30 @@
+      border-top-right-radius: rem-calc(6);
+      margin-bottom: 0;
+      margin-right: $line-height / 4;
+      margin-top: 0;
```
These styles are already in the original file, right? Can we remove them from here, then, or am I missing something? :thinking:

javierm commented on app/assets/stylesheets/custom/legislation_process.scss:
```diff
@@ -0,0 +1,30 @@
+        span {
+          color: $white;
+        }
+      }
```
IMHO it's OK to leave the original Consul color here as well, since we're using that color on that very same page.

javierm commented on app/assets/stylesheets/custom/layout.scss:
```diff
@@ -0,0 +1,63 @@
+    &:hover {
+      color: $brand;
+    }
+  }
```
This generates:

```css
footer a:hover,
footer a:active:hover,
footer a:focus:hover {
  color: #ba283b;
}
```

Not sure what we mean here :thinking:.

javierm reviewed (commented):
@taitus I've left some comments (and so has Pronto!); let me know what you think :wink:.

javierm committed:
Fix menu on small screens

The aspect ratio of the header logo is 1.6:1 instead of the 3.25:1 which
is the default one in Consul.

This made the menu look broken on small screens when users aren't
identified. While this has been fixed in Consul's master branch commit
045ac648d, we are increasing the size of the logo as a workaround until
Consul 2.0.0 is released and we upgrade to that version.
